core/02_error.js

```diff
@@ -1,157 +1,158 @@
+    evalOrigin: callSite.getEvalOrigin(),
+    isToplevel: callSite.isToplevel(),
+    isEval: callSite.isEval(),
+    isNative: callSite.isNative(),
+    isConstructor: callSite.isConstructor(),
+    isAsync: callSite.isAsync(),
+    isPromiseAll: callSite.isPromiseAll(),
+    promiseIndex: callSite.getPromiseIndex(),
+  };
+}
+
+function sourceMapCallSiteEval(cse) {
+  if (cse.fileName && cse.lineNumber != null && cse.columnNumber != null) {
+    return { ...cse, ...ops.op_apply_source_map(cse) };
   }
+  return cse;
+}
 
-  /** A function that can be used as `Error.prepareStackTrace`. */
-  function prepareStackTrace(error, callSites) {
-    let callSiteEvals = ArrayPrototypeMap(callSites, evaluateCallSite);
-    callSiteEvals = ArrayPrototypeMap(callSiteEvals, sourceMapCallSiteEval);
-    ObjectDefineProperties(error, {
-      __callSiteEvals: { __proto__: null, value: [], configurable: true },
-    });
-    const formattedCallSites = [];
-    for (let i = 0; i < callSiteEvals.length; ++i) {
-      const cse = callSiteEvals[i];
-      ArrayPrototypePush(error.__callSiteEvals, cse);
-      ArrayPrototypePush(formattedCallSites, formatCallSiteEval(cse));
-    }
-    const message = error.message !== undefined ? error.message : "";
-    const name = error.name !== undefined ? error.name : "Error";
-    let messageLine;
-    if (name != "" && message != "") {
-      messageLine = `${name}: ${message}`;
-    } else if ((name || message) != "") {
-      messageLine = name || message;
-    } else {
-      messageLine = "";
-    }
-    return messageLine +
-      ArrayPrototypeJoin(
-        ArrayPrototypeMap(formattedCallSites, (s) => `\n    at ${s}`),
-        "",
-      );
+/** A function that can be used as `Error.prepareStackTrace`. */
+function prepareStackTrace(error, callSites) {
+  let callSiteEvals = ArrayPrototypeMap(callSites, evaluateCallSite);
+  callSiteEvals = ArrayPrototypeMap(callSiteEvals, sourceMapCallSiteEval);
+  ObjectDefineProperties(error, {
+    __callSiteEvals: { __proto__: null, value: [], configurable: true },
+  });
+  const formattedCallSites = [];
+  for (let i = 0; i < callSiteEvals.length; ++i) {
+    const cse = callSiteEvals[i];
+    ArrayPrototypePush(error.__callSiteEvals, cse);
+    ArrayPrototypePush(formattedCallSites, formatCallSiteEval(cse));
+  }
+  const message = error.message !== undefined ? error.message : "";
+  const name = error.name !== undefined ? error.name : "Error";
+  let messageLine;
+  if (name != "" && message != "") {
+    messageLine = `${name}: ${message}`;
+  } else if ((name || message) != "") {
+    messageLine = name || message;
+  } else {
+    messageLine = "";
   }
+  return messageLine +
+    ArrayPrototypeJoin(
+      ArrayPrototypeMap(formattedCallSites, (s) => `\n    at ${s}`),
+      "",
+    );
+}
 
-  ObjectAssign(globalThis.__bootstrap.core, { prepareStackTrace });
-  ObjectFreeze(globalThis.__bootstrap.core);
-})(this);
+ObjectAssign(core, { prepareStackTrace });
```

Comment:
Though it was originally like this, does it actually make any sense to use `ObejctAssign` instead of just plain assigning keys to the core object?

Reply:
yea probably could be cleaned up

Reply:
done